Modules/_sqlite/connection.c

```diff
@@ -1478,6 +1478,92 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
 }
 
 static PyObject *
+pysqlite_connection_backup(pysqlite_Connection* self, PyObject* args, PyObject* kwds)
+{
+    char* filename;
+    int pages = -1;
+    PyObject* progress = Py_None;
+    PyObject* retval = NULL;
+    int rc;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"filename", "pages", "progress", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "s|$iO:backup", keywords,
+                                     &filename, &pages, &progress)) {
+        goto finally;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
+        PyErr_SetString(PyExc_TypeError, "progress argument must be a callable");
+        goto finally;
+    }
+
+    if (pages == 0) {
+        pages = -1;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    rc = sqlite3_open(filename, &bckconn);
+    Py_END_ALLOW_THREADS
+
+    if (rc != SQLITE_OK) {
+        goto finally;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    bckhandle = sqlite3_backup_init(bckconn, "main", self->db, "main");
```

Comment:
That's not really the _database name_: accordingly to the [documentation](http://sqlite.org/c3ref/backup_finish.html#sqlite3backupinit) that name denotes the target, that can either be "main" for the main storage, "temp" for the temporary storage or the name of the _attached_ database.
Of course we can easily expose that as a (default) argument, but I did not see as necessary when I coded the feature.